components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -7454,15 +7458,26 @@ def getChannels(self):
         # E.g. ConcurrencyException (no rendering engine): load channels by
         # hand, use pixels to order channels
         else:
-            pid = self.getPixelsId()
-            params = omero.sys.Parameters()
-            params.map = {"pid": rlong(pid)}
-            query = ("select p from Pixels p join fetch p.channels as c "
-                     "join fetch c.logicalChannel as lc where p.id=:pid")
-            pixels = self._conn.getQueryService().findByQuery(
-                query, params, self._conn.SERVICE_OPTS)
-            return [ChannelWrapper(self._conn, c, idx=n, re=self._re, img=self)
-                    for n, c in enumerate(pixels.iterateChannels())]
+            return self.getChannelsNoRE()
+
+    def getChannelsNoRE(self):
```

Comment:
Is there a reason to have a new method here but a new parameter on `channelMetadata`? My gut feeling would be to do the same in both places.